package.json

```diff
@@ -29,12 +29,13 @@
     "deep-equal": "1.0.1",
     "invariant": "2.2.0",
     "react-side-effect": "1.0.2",
+    "recompose": "^0.14.5",
```

Comment:
I agree with @srph and feel like we should accomplish this without adding another dependency to the library.  @evenchange4 - Thank you for your PR, if you'd like to give it another try without adding a dependency, I'll definitely take a look.